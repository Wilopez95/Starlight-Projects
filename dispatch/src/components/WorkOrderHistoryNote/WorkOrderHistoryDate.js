import moment from 'moment';
import PropTypes from 'prop-types';

const WorkOrderHistoryDate = (props) => {
  const { createdDate } = props;
  return <p>{moment(createdDate).format('MM/DD/YYYY')}</p>;
};

WorkOrderHistoryDate.propTypes = {
  createdDate: PropTypes.string.isRequired,
};
export default WorkOrderHistoryDate;
