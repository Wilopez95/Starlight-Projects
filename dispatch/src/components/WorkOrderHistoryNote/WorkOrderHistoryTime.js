import moment from 'moment-timezone';
import PropTypes from 'prop-types';

// export type Props = {
//   createdDate: string,
//   timezone: string,
// };

const WorkOrderHistoryTime = (props) => {
  const timez = props.timezone || Intl.DateTimeFormat().resolvedOptions().timeZone;
  return <p>{moment.utc(props.createdDate).tz(timez).format('hh:mm a')}</p>;
};

WorkOrderHistoryTime.defaultProps = {
  timezone: 'America/Denver',
};

WorkOrderHistoryTime.propTypes = {
  createdDate: PropTypes.string.isRequired,
  timezone: PropTypes.string,
};
export default WorkOrderHistoryTime;
