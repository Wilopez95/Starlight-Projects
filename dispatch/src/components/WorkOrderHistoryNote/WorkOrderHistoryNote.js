/* eslint-disable react/prop-types */
/* eslint-disable complexity */

import cx from 'classnames';
import WorkOrderHistoryTime from './WorkOrderHistoryTime';
import WorkOrderHistoryDate from './WorkOrderHistoryDate';

// type Props = {
//   data: Object,
//   notesStates: Object,
//   timezone: string,
// };

const WorkOrderHistoryNote = (props) => {
  const {
    data,
    data: { note, location },
    notesStates,
    timezone,
  } = props;

  const isTransition = data.type === notesStates.TRANSITION;
  const title = isTransition ? data.note.newState : data.type;

  const noteClassNameByType = {
    [notesStates.NOTE]: 'note--default',
    [notesStates.MANIFEST]: 'note--manifest',
    [notesStates.SCALETICKET]: 'note--scaleTickets',
    [notesStates.TRANSITION]: 'note--transition',
    WEIGHT_RECORD: 'note--weight',
  };

  return (
    <div className={cx('note', noteClassNameByType[data.type])}>
      <div className="note-column note-column--date">
        {data.createdDate ? <WorkOrderHistoryDate createdDate={data.createdDate} /> : null}
        {data.createdDate ? (
          <WorkOrderHistoryTime createdDate={data.createdDate} timezone={timezone} />
        ) : null}
      </div>
      <div className="note-column note-column--author">
        {data.createdBy ? <p>{data.createdBy}</p> : null}
      </div>
      <div className="note-column note-column--title">
        <p>{title ? String(title).replace(/_/g, ' ') : 'TRANSITION'}</p>
      </div>
      <div className="note-column note-column--details">
        <ul className="details-list">
          {note.text ||
          note.weight ||
          note.quantity ||
          note.picture ||
          note.document ||
          location.name ||
          location.waypointName ||
          (location.location && location.location.lat) ? (
            <li className="details-item">
              {note.text ? <p>{note.text}</p> : null}
              {note.weight ? (
                <p>
                  Weight: {note.weight} {note.unittype ? <span>{note.unittype}</span> : null}
                </p>
              ) : null}
              {note.quantity ? (
                <p>
                  {note.quantity} {note.unittype ? <span>{note.unittype}</span> : 'UNITS'}
                </p>
              ) : null}
              {note.picture ? (
                <p>
                  File Attached:{' '}
                  <a href={note.picture} rel="noopener noreferrer" target="_blank">
                    Link
                  </a>
                </p>
              ) : null}
              {note.document ? (
                <p>
                  Signed Document:{' '}
                  <a href={note.document} rel="noopener noreferrer" target="_blank">
                    Link
                  </a>
                </p>
              ) : null}
              <p>
                {/^GPS_/.test(location.name)
                  ? `Lat: ${location.location.lat} Long: ${location.location.lon}`
                  : location.waypointName || location.name}
              </p>
            </li>
          ) : null}
          {data.can && data.can.name ? (
            <li className="details-item">
              <p>
                <b>CAN:</b> {data.can.name}
              </p>
            </li>
          ) : null}
          {note.manifestNumber || note.profileNumber || note.ticketNumber ? (
            <li className="details-item details-item--numbers">
              {note.manifestNumber ? <p>MANIFEST: #{note.manifestNumber}</p> : null}
              {note.profileNumber ? <p>PROFILE: #{note.profileNumber}</p> : null}
              {note.ticketNumber ? <p>Ticket: #{note.ticketNumber}</p> : null}
            </li>
          ) : null}
        </ul>
      </div>
    </div>
  );
};

WorkOrderHistoryNote.defaultProps = {
  timezone: 'America/Denver',
};

export default WorkOrderHistoryNote;
