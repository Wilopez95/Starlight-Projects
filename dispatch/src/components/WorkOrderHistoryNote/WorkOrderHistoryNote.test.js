import { shallow, mount } from 'enzyme';
import WorkOrderHistoryNote from './WorkOrderHistoryNote';

const timezone = 'America/Denver';
const notesStates = {
  NOTE: 'NOTE',
  MANIFEST: 'MANIFEST',
  SCALETICKET: 'SCALETICKET',
  TRANSITION: 'TRANSITION',
};
const data = {
  location: {
    id: null,
    name: null,
    description: null,
    type: null,
    waypointType: null,
    waypointName: null,
    location: {
      lon: null,
      lat: null,
    },
    createdBy: null,
    createdDate: null,
    modifiedBy: null,
    modifiedDate: null,
  },
  can: {
    id: null,
    name: null,
    serial: null,
    size: null,
    requiresMaintenance: null,
    outOfService: null,
    source: null,
    startDate: null,
    hazardous: null,
    action: null,
    timestamp: null,
    createdBy: null,
    createdDate: null,
    modifiedBy: null,
    modifiedDate: null,
  },
  id: 2056967,
  workOrderId: 200486,
  type: 'TRANSITION',
  note: {
    text: 'Reassigned from Bert Rodriguez Rt13 to Manuel Berlin Rt 24',
    driverId: 65,
    newState: 'REASSIGNMENT',
  },
  createdBy: 'alycia.hilzer@5280waste.com',
  createdDate: '2019-08-26T00:53:45.680Z',
  modifiedBy: null,
  modifiedDate: '2019-08-26T00:53:45.680Z',
  deleted: 0,
};
const dataNote = {
  location: {
    id: null,
    name: null,
    description: null,
    type: null,
    waypointType: null,
    waypointName: null,
    location: {
      lon: null,
      lat: null,
    },
    createdBy: null,
    createdDate: null,
    modifiedBy: null,
    modifiedDate: null,
  },
  can: {
    id: null,
    name: null,
    serial: null,
    size: null,
    requiresMaintenance: null,
    outOfService: null,
    source: null,
    startDate: null,
    hazardous: null,
    action: null,
    timestamp: null,
    createdBy: null,
    createdDate: null,
    modifiedBy: null,
    modifiedDate: null,
  },
  id: 2056967,
  workOrderId: 200486,
  type: 'NOTE',
  note: {
    text: 'Reassigned from Bert Rodriguez Rt13 to Manuel Berlin Rt 24',
    driverId: 65,
    newState: 'REASSIGNMENT',
  },
  createdBy: 'alycia.hilzer@5280waste.com',
  createdDate: '2019-08-26T00:53:45.680Z',
  modifiedBy: null,
  modifiedDate: '2019-08-26T00:53:45.680Z',
  deleted: 0,
};

describe('<WorkOrderHistoryNote />', () => {
  it('should render correctly with props', () => {
    const component = shallow(
      <WorkOrderHistoryNote
        data={data}
        timezone={timezone}
        notesStates={notesStates}
      />,
    );

    expect(component).toMatchSnapshot();
  });
  it('should have the classname note--transition added for a TRANSITION type by noteClassNameByType', () => {
    const wrapper = mount(
      <WorkOrderHistoryNote
        data={data}
        timezone={timezone}
        notesStates={notesStates}
      />,
    );
    expect(wrapper.find('.note').hasClass('note--transition')).toEqual(true);
  });
  it('should have the classname note--default added for a NOTE type by noteClassNameByType', () => {
    const wrapper = mount(
      <WorkOrderHistoryNote
        data={dataNote}
        timezone={timezone}
        notesStates={notesStates}
      />,
    );
    expect(wrapper.find('.note').hasClass('note--default')).toEqual(true);
  });
});
