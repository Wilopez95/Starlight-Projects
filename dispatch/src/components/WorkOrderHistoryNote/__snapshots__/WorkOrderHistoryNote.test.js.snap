// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkOrderHistoryNote /> should render correctly with props 1`] = `ShallowWrapper {}`;
